### PR TITLE
Add KVUsage to consul state usage metrics

### PR DESCRIPTION
**Pull request by clly**

### CODE REVIEW

**Review by karl-cardenas-coding — APPROVED** [review]
Good from a docs perspective

File: agent/consul/state/usage_test.go [hunk 1]
```diff
@@ -45,6 +45,44 @@ func TestStateStore_Usage_NodeUsage_Delete(t *testing.T) {
 	require.Equal(t, usage.Nodes, 1)
 }
 
+func TestStateStore_Usage_KVUsage(t *testing.T) {
+	s := testStateStore(t)
+
+	// No nodes have been registered, and thus no usage entry exists
```
Comment (mkeeler) [inline agent/consul/state/usage_test.go:51]:
```suggestion
	// No keys have been registered, and thus no usage entry exists
```

File: agent/consul/usagemetrics/usagemetrics.go [hunk 2]
```diff
@@ -145,6 +149,7 @@ func (u *UsageMetricsReporter) Run(ctx context.Context) {
 }
 
 func (u *UsageMetricsReporter) runOnce() {
+	u.logger.Debug("Starting usage run")
```
Comment (mkeeler) [inline agent/consul/usagemetrics/usagemetrics.go:152]:
```suggestion
	u.logger.Trace("Starting usage run")
```

Do you think trace level would be more appropriate than debug here?

File: agent/consul/usagemetrics/usagemetrics.go [hunk 3]
```diff
@@ -163,6 +168,14 @@ func (u *UsageMetricsReporter) runOnce() {
 
 	members := u.memberUsage()
 	u.emitMemberUsage(members)
+
+	_, kvUsage, err := state.KVUsage()
+	if err != nil {
+		u.logger.Warn("failed to retrieve kv entries from state store", "error", err)
```
Comment (mkeeler) [inline agent/consul/usagemetrics/usagemetrics.go:174]:
```suggestion
		u.logger.Warn("failed to retrieve kv entry usage from state store", "error", err)
```

**Review by mkeeler — APPROVED** [review]
This looks great. I left a few comments but they are ultra-nitpicky.

File: agent/consul/usagemetrics/usagemetrics.go (hunk 2 above)
Comment (clly) [reply agent/consul/usagemetrics/usagemetrics.go:152]: No strong opinions

File: agent/consul/state/usage_oss.go [hunk 4]
```diff
@@ -1,3 +1,4 @@
+//go:build !consulent
```
Comment (rboyer) [inline agent/consul/state/usage_oss.go:1]: Consul is still living in Go `1.16.7` so I don't think this needs to be here yet.

Comment (clly) [reply agent/consul/state/usage_oss.go:1]: we're running go `1.17` and vscode probably inserted it automatically. I can remove it if you prefer to leave it out but it shouldn't break anything should it?

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/453132.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 1e3ba26223420f5508250d18e7ccaff852be380d onto `release/1.11.x` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 1e3ba26223420f5508250d18e7ccaff852be380d onto `release/1.10.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/453132)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 1e3ba26223420f5508250d18e7ccaff852be380d onto `release/1.9.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/453132)

File: agent/consul/usagemetrics/usagemetrics.go [hunk 5]
```diff
@@ -36,6 +36,10 @@ var Gauges = []prometheus.GaugeDefinition{
 		Name: []string{"consul", "members", "servers"},
 		Help: "Measures the current number of server agents registered with Consul. It is only emitted by Consul servers. Added in v1.9.6.",
 	},
+	{
+		Name: []string{"consul", "kv", "entries"},
+		Help: "Measures the current number of server agents registered with Consul. It is only emitted by Consul servers. Added in v1.10.3.",
```
Comment (commarla) [inline agent/consul/usagemetrics/usagemetrics.go:41]: Should be `Measures the current number of unique KV entries written in Consul. It is only emitted by Consul servers. Added in v1.10.3.`

**Pull request merged**